jOOL-java-8/src/main/java/org/jooq/lambda/Agg.java

```diff
@@ -979,4 +976,149 @@ else if (percentile == 1.0)
             s -> s.map(Objects::toString).orElse("")
         );
     }
+
+
+    //CS304 Issue link: https://github.com/jOOQ/jOOL/issues/360
+
+    /**
+     * Calculate the variance of objects with mapping function from object to double value,
+     * with given list, size and mapping function.
+     *
+     * <p><pre>
+     *     This function is based on the definition of standard deviation:
+     *     First, calculate the sum of value of all objects and the average value;
+     *     Second, calculate the sum of square of difference between each objects and average;
+     *     Third, use sum of variance divide size to obtain variance of all objects.
+     * </pre>
+     *
+     * @param function mapping function from object to double value
+     * @param l        a list containing all the objects
+     * @param size     the size of the list in collector
+     * @return the variance value
+     * @version 1.0
+     * @author Jichen Lu
+     * @date 2021-04-25
+     */
+    private static <T> double getVariance(
+            Function<? super T, Double> function,
+            ArrayList<T> l, int size) {
+        double sum = 0.0;
+        double average;
+        double sumVariance = 0.0;
+        double variance;
+        for (T o : l) {
+            double temp = function.apply(o);
+            sum += temp;
+        }
+        average = sum / size;
+        for (T o : l) {
+            double temp = Math.pow(function.apply(o) - average, 2);
+            sumVariance += temp;
+        }
+        variance = sumVariance / size;
+        return variance;
+    }
+
+
+    //CS304 Issue link: https://github.com/jOOQ/jOOL/issues/360
+
+    /**
+     * Calculate the variance of the given object collectors,
+     * based on the mapping function from object to double number.
+     *
+     * <p><pre> Usage of aggregation function stddevBy():
+     * The mapping function is a function mapping the objects to a double value, for instance:
+     * {@code
+     * Function<Integer, Double> mapping = e -> Double.valueOf(e);
+     * }
+     * The specific usage of stddevBy is like:
+     * {@code
+     * Seq.of(1, 1, 1, 1).collect(Agg.stddevBy(mapping));
+     * }
+     * Besides, self defined class is also allowed with mapping function:
+     * {@code
+     * Seq.of(new Node(1), new Node(1), new Node(1), new Node(1)).collect(Agg.stddevBy(mapping1));
+     * }
+     * </pre>
+     *
+     * @param function mapping function from object to double value
+     * @return the stddev value
+     * @version 1.0
+     * @author Jichen Lu
+     * @date 2021-04-25
+     */
+    public static <T> Collector<T, ?, Optional<Double>> stddevBy(
+            Function<? super T,
+                    Double> function) {
```

Comment:
While I aprreciate that there can be an optimised implementation for `double` and `float`, I think this is missing the point of jOOL to be `Number` subtype agnostic in such cases. See e.g. `Agg.sum()`, which can sum any `Number` subtype with specialised implementations in `org.jooq.lambda.Sum`

Reply:
I've given this a bit more thought. I don't think that statistical functions really make sense with integer types, so we can skip those. Also, we have a precedent where we're offering only `Optional<Double>` as a collector result: `percentRank()`.

Reply:
OK, an even better precedent. Following the `Collectors.summingInt()` etc. naming convention, we have a `Agg.bitOr()` collector that is generic, and a `Agg.bitOrInt()` that is primitive. That'll do. Your contribution will be stddevDouble